File: wk7/tut/word_frequency.pl
#!/usr/bin/perl -w

%count_words = ();

while ($line = <STDIN>) {
    $line = lc $line;
    @words = split /[^a-z]+/, $line;
    foreach $word (@words) {
        $count_words{$word} += 1;
    }
}

sub by_count {
    $count_words{$a} <=> $count_words{$b};
}

foreach $key (sort by_count (keys %count_words)) {
    print "$count_words{$key} $key\n";
}



$words{$prev}{$current} += 1;

$words{$prev}

File: wk7/tut/word_frequency.py
#!/usr/bin/env python3
import sys
import re
from collections import defaultdict

freq = defaultdict(int)
lines = sys.stdin.readlines()
s = ''.join(lines).lower()

for word in re.split(r'\W+', s):
    freq[word] += 1

for word in sorted(freq, key=freq.get):
    print('{} {}'.format(freq[word], word))
